sw/device/tests/sim_dv/csrng_lc_hw_debug_en_test.c

```diff
@@ -0,0 +1,280 @@
+            0x02,
+            0x03,
+            0x04,
+            0x05,
+            0x06,
+            0x07,
+            0x08,
+            0x09,
+            0x0a,
+            0x0b,
+            0x0c,
+            0x0d,
+            0x0e,
+            0x0f,
+        },
+};
+
+/**
+ * Initialize the peripherals used in this test.
+ */
+static void init_peripherals(void) {
+  CHECK_DIF_OK(dif_csrng_init(
+      mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR), &csrng));
+  CHECK_DIF_OK(dif_lc_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR), &lc_ctrl));
+  CHECK_DIF_OK(dif_entropy_src_init(
+      mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR), &entropy_src));
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp_ctrl));
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+
+  CHECK_DIF_OK(dif_flash_ctrl_init_state(
+      &flash_ctrl_state,
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+}
+
+static void lock_otp_secret0_partition(void) {
+  uint64_t otp_exit_token_0 = 0;
+  uint64_t otp_exit_token_1 = 0;
+  const size_t kTokenSize = ARRAYSIZE(kLcExitToken.data);
+  for (size_t i = 0; i < kTokenSize; i++) {
+    if (i < kTokenSize / 2) {
+      otp_exit_token_0 |= (uint64_t)kLcExitToken.data[i] << (i * 8);
+    } else {
+      otp_exit_token_1 |= (uint64_t)kLcExitToken.data[i]
+                          << ((i - kTokenSize / 2) * 8);
+    }
+  }
+  CHECK_DIF_OK(dif_otp_ctrl_dai_program64(&otp_ctrl,
+                                          kDifOtpCtrlPartitionSecret0,
+                                          /*address=*/0x10,
+                                          /*value=*/otp_exit_token_0));
+  otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+  CHECK_DIF_OK(dif_otp_ctrl_dai_program64(&otp_ctrl,
+                                          kDifOtpCtrlPartitionSecret0,
+                                          /*address=*/0x18,
+                                          /*value=*/otp_exit_token_1));
+  otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+
+  CHECK_DIF_OK(dif_otp_ctrl_dai_digest(&otp_ctrl, kDifOtpCtrlPartitionSecret0,
+                                       /*digest=*/0));
+  otp_ctrl_testutils_wait_for_dai(&otp_ctrl);
+}
+
+static dif_lc_ctrl_state_t lc_next_state(void) {
+  uint32_t index = rand_testutils_gen32_range(/*min=*/0, /*max=*/2);
+  dif_lc_ctrl_state_t next_state;
+  switch (index) {
+    case 0:
+      next_state = kDifLcCtrlStateDev;
+      break;
+    case 1:
+      next_state = kDifLcCtrlStateProd;
+      break;
+    case 2:
+      next_state = kDifLcCtrlStateProdEnd;
+      break;
+    default:
+      CHECK(false, "Unexpected case index: %d", index);
+      break;
+  }
+  return next_state;
+}
+
+static void entropy_conditioner_stop(const dif_entropy_src_t *entropy_src) {
+  uint32_t fail_count = 0;
+  dif_result_t op_result;
+  do {
```

Comment:
i don't completely understand this part, does this mean the conditioner stop does not always work? 
I thought @martin-lueker had changed some of this stuff to..."will stop eventually, but perhaps not immediately".

Let me know if I'm misunderstanding.

Reply:
o wait i think i get it now. So this thing doesn't allow you to stop until the fifos have been consumed.

Reply:
That is correct.   If you try to stuff data in or turn it off before the data has been absorbed it will fail.

Reply:
And you probably will not get the output you want.  The new delays in the system add some uncertainty to this process, but before (when every thing happened instantly) this would definitely lead to the wrong output.

Reply:
sounds good, thanks for explaining @martin-lueker